### ISSUE TITLE
Improve visual styling of grid drawing

### ISSUE THREAD

**Issue by quaquel** [projectmesa/mesa#2438]
Grids are optionally drawn for all discrete spaces. However, the look is not always esthetically pleasing. It might be worth to spend some time to tune how grids are drawn for all spaces.

**Comment by AdeebaNizam404** [comment]
Can u please assign this  issue to me ..... I want to work on it ....

**Comment by EwoutH** [comment]
Thanks for showing interest, @quaquel good by you?

**Comment by quaquel** [comment]
Yes go ahead and put forward your ideas. Note that it should be done in a way that does not break the existing API, although having more user control (so expanding the API) would be fine. Moreover, even if you were to start on the matplotlib side, it is important to have a sense of how it might work for altair as well.

**Comment by Sahil-Chhoker** [comment]
@quaquel I would like to work on this issue, but I don't understand what's wrong with the current grids. Do you mind explaining what's not so esthetically pleasing about them?

**Comment by quaquel** [comment]
When visualizing discrete spaces, hashed lines are used by default to draw the grid. See, any of the advanced examples for an illustration. Moreover, in particular in case of hexgrids, the way this grid is created is problematic because it results in overplotting. Again, create a hexgrid and visualize it using matplotlib to get a sense of what I mean.

**Comment by Sahil-Chhoker** [comment]
Should the function allow changing the style of the old dotted line using a single parameter, or should it support customizing various parameters such as line width, color, or linestyle? Alternatively, should it include automatic functionality, like adjusting the line width based on the distance between two lines (though it's unclear if this is feasible)?

**Comment by quaquel** [comment]
Anything goes, and ideas are welcome. However, please see my earlier remarks above.

My hunch is to start simple: solve the hex tiling overplotting issue first. 

My longer-term vision, in light also of #2557 and #2437, is to separate explicitly the drawing of the space itself from the drawing of agents within the space. Drawing the space itself involves support for property layers and support for any gridding. Agent drawing is well covered already.
